manifests/iaas/azure.yml

```diff
@@ -4,26 +4,42 @@ params:
   azure_resource_group: (( param "Specify the resource group for Azure" ))
   azure_default_sg:     (( param "Specify the default security group for your Azure VMs" ))
 
+meta:
+  director_cpi_job: azure_cpi
+
 instance_groups:
 - name: bosh
   jobs:
   - name: azure_cpi
     release: bosh-azure-cpi
-  properties:
-    director:
-      cpi_job: azure_cpi
+    properties:
+      azure:
+        environment: (( grab params.azure_environment ))
+        subscription_id: (( vault meta.vault "/azure:subscription_id" ))
+        tenant_id: (( vault meta.vault "/azure:tenant_id" ))
+        client_id: (( vault meta.vault "/azure:client_id" ))
+        client_secret: (( vault meta.vault "/azure:client_secret" ))
+        resource_group_name: (( grab params.azure_resource_group ))
+        default_security_group: (( grab params.azure_default_sg ))
+        ssh_user: vcap
+        ssh_public_key: (( vault meta.vault "/azure/ssh:public" ))
+        use_managed_disks: true
+      blobstore:
+        .: (( inject instance_groups.bosh.jobs.blobstore.properties.blobstore ))
```

Comment:
This is just a grab...

Reply:
Don't grab hashes in genesis kits. It causes unexpected behavior when overriding individual values in the env files...